sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs

```diff
@@ -33,22 +32,73 @@
+         * - loads a different solution in the IDE
+         *
+         * If caching needs to be done in the future, it should take into account:
+         * - diffing the contents of the configuration file
+         * - associating the file with a unique identifier for the build project
+         */
+        internal static void SetParameterValues(ParameterLoadingDiagnosticAnalyzer parameteredAnalyzer,
+            AnalyzerOptions options)
         {
-            public string ParameterKey { get; set; }
+            var additionalFile = options.AdditionalFiles.FirstOrDefault(f => IsSonarLintXml(f.Path));
 
-            public string ParameterValue { get; set; }
+            if (additionalFile == null)
+            {
+                return;
+            }
+
+            ImmutableList<RuleParameterValues> parameters;
+            try
+            {
+                using var xtr = XmlReader.Create(additionalFile.Path);
+                var xml = XDocument.Load(xtr);
+                parameters = ParseParameters(xml);
+            }
+            catch (Exception ex) when (ex is IOException || ex is XmlException)
+            {
+                // cannot log exception
+                return;
+            }
+
+            var propertyParameterPairs = parameteredAnalyzer.GetType()
+                .GetRuntimeProperties()
+                .Select(p => new { Property = p, Descriptor = p.GetCustomAttributes<RuleParameterAttribute>().SingleOrDefault() })
+                .Where(p => p.Descriptor != null);
+
+            var ids = new HashSet<string>(parameteredAnalyzer.SupportedDiagnostics.Select(diagnostic => diagnostic.Id));
+            foreach (var propertyParameterPair in propertyParameterPairs)
+            {
+                var parameter = parameters
+                    .FirstOrDefault(p => ids.Contains(p.RuleId));
+
+                var parameterValue = parameter?.ParameterValues
+                    .FirstOrDefault(pv => pv.ParameterKey == propertyParameterPair.Descriptor.Key);
+
+                if (TryConvertToParameterType(parameterValue?.ParameterValue, propertyParameterPair.Descriptor.Type, out var value))
+                {
+                    propertyParameterPair.Property.SetValue(parameteredAnalyzer, value);
+                }
+            }
         }
 
-        private class RuleParameterValues
+        internal static bool IsSonarLintXml(string path)
         {
-            public string RuleId { get; set; }
+            return ConfigurationFilePathMatchesExpected(path, ParameterConfigurationFileName);
+        }
 
-            public List<RuleParameterValue> ParameterValues { get; } = new List<RuleParameterValue>();
+        internal static bool ConfigurationFilePathMatchesExpected(string path, string fileName)
+        {
+            return new FileInfo(path).Name.Equals(fileName, StringComparison.OrdinalIgnoreCase);
         }
 
         private static ImmutableList<RuleParameterValues> ParseParameters(XDocument xml)
```

Comment:
I changed it, although for private methods I don't think it's that important

also, for `SetParameterValues` I want to keep `ParameterLoadingDiagnosticAnalyzer` because this class should be used by that class (it's kinda tightly coupled)